lightning/src/ln/outbound_payment.rs

```diff
@@ -902,15 +902,64 @@ impl OutboundPayments {
 			}
 		}
 
-		let mut route = router.find_route_with_id(
-			&node_signer.get_node_id(Recipient::Node).unwrap(), &route_params,
+		let payer = node_signer.get_node_id(Recipient::Node).unwrap();
+		let mut route;
+		match router.find_route_with_id(
+			&payer,&route_params,
 			Some(&first_hops.iter().collect::<Vec<_>>()), inflight_htlcs(),
 			payment_hash, payment_id,
-		).map_err(|_| {
-			log_error!(logger, "Failed to find route for payment with id {} and hash {}",
-				payment_id, payment_hash);
-			RetryableSendFailure::RouteNotFound
-		})?;
+		) {
+			Ok(res) => route = res,
+			Err(error) => {
+				if error.err == "Cannot generate a route to ourselves" {
+					let payment_secret = match recipient_onion.payment_secret {
+						Some(secret) => secret,
+						None => PaymentSecret([0; 32])
```

Comment:
Not sure this should ever be set to a dummy value. @TheBlueMatt Do we want to `inbound_payment::verify` the payment secret here? I'd think that's how we're mitigating the self-payment attack described on the ML.

Reply:
@TheBlueMatt  w/r/t treating this like any other payment, do we want to give users the opportunity to fail this payment? cc https://github.com/lightningdevkit/rust-lightning/pull/2573#discussion_r1394458370

Reply:
@vladimirfomene I think senders still don't have the opportunity to fail these payments, it looks like they're always claimed immediately?